ext/reflection/php_reflection.c

```diff
@@ -6248,12 +6253,16 @@ ZEND_METHOD(ReflectionProperty, skipLazyInitialization)
 		RETURN_THROWS();
 	}
 
-	bool prop_was_lazy = (Z_PROP_FLAG_P(OBJ_PROP(object, ref->prop->offset)) & IS_PROP_LAZY);
+	while (zend_object_is_lazy_proxy(object)
+			&& zend_lazy_object_initialized(object)) {
+		object = zend_lazy_object_get_instance(object);
+	}
 
 	zval *src = &object->ce->default_properties_table[OBJ_PROP_TO_NUM(ref->prop->offset)];
 	zval *dst = OBJ_PROP(object, ref->prop->offset);
+	bool prop_was_lazy = (Z_PROP_FLAG_P(dst) & IS_PROP_LAZY);
 
-	if (!(Z_PROP_FLAG_P(dst) & IS_PROP_LAZY)) {
+	if (!prop_was_lazy) {
```

Comment:
`prop_was_lazy` is used below, and always `true`. Is this correct? In that case, it should be dropped from the condition.